hack/update-codegen.sh

```diff
@@ -23,7 +23,7 @@ SCRIPT_ROOT=$(dirname "${BASH_SOURCE[@]}")/..
 TOOLS_DIR=$(realpath ./hack/tools)
 TOOLS_BIN_DIR="${TOOLS_DIR}/bin"
 GO_INSTALL=$(realpath ./hack/go-install.sh)
-CONTROLLER_GEN_VER=v0.16.5
+CONTROLLER_GEN_VER=v0.18.0
```

Comment:
v0.17.3 is the exact match: https://github.com/kubernetes-sigs/controller-tools/blob/v0.17.3/go.mod#L16